lib/net.c

```diff
@@ -1842,6 +1842,10 @@ finalize_stage2(void *arg)
 		}
 		return drop_privileges();
 	}
+
+	/* Enable rate-limited logging now that startup is complete. */
+	log_ratelimit_enable();
+
```

Comment:
If the execution reaches line `return drop_privileges();`, `log_ratelimit_enable()` is never called. We need to treat the return of `drop_privileges()`.